barretenberg/cpp/src/barretenberg/stdlib/primitives/biggroup/biggroup_impl.hpp

```diff
@@ -524,104 +524,164 @@ element<C, Fq, Fr, G> element<C, Fq, Fr, G>::multiple_montgomery_ladder(
         bool is_negative = false;
     };
 
-    Fq previous_x = _x;
-    composite_y previous_y{ std::vector<Fq>(), std::vector<Fq>(), std::vector<Fq>(), false };
-    for (size_t i = 0; i < add.size(); ++i) {
+    // Handle edge case of empty input
+    if (add.empty()) {
+        return *this;
+    }
+
+    // Let A = (x, y) and P = (x₁, y₁)
+    // For the first point P, we want to compute: (2A + P) = (A + P) + A
+    // We first need to check if x ≠ x₁.
+    x().assert_is_not_equal(add[0].x3_prev);
+
+    // Compute λ₁ for computing the first addition: (A + P)
+    Fq lambda1;
+    if (!add[0].is_element) {
+        // Case 1: P is an accumulator (i.e., it lacks a y-coordinate)
+        //         λ₁ = (y - y₁) / (x - x₁)
+        //            = -(y₁ - y) / (x - x₁)
+        //            = -(λ₁_ₚᵣₑᵥ * (x₁_ₚᵣₑᵥ - x₁) - y₁_ₚᵣₑᵥ - y) / (x - x₁)
+        //
+        // NOTE: msub_div computes -(∑ᵢ aᵢ * bᵢ + ∑ⱼcⱼ) / d
+        lambda1 = Fq::msub_div({ add[0].lambda_prev },              // numerator left multiplicands: λ₁_ₚᵣₑᵥ
+                               { add[0].x1_prev - add[0].x3_prev }, // numerator right multiplicands: (x₁_ₚᵣₑᵥ - x₁)
+                               (x() - add[0].x3_prev),              // denominator: (x - x₁)
+                               { -add[0].y1_prev, -y() },           // numerator additions: -y₁_ₚᵣₑᵥ - y
+                               /*enable_divisor_nz_check*/ false);  // divisor check is not needed as x ≠ x₁ is enforced
+    } else {
+        // Case 2: P is a full element (i.e., it has a y-coordinate)
+        //         λ₁ = (y - y₁) / (x - x₁)
+        //
+        lambda1 = Fq::div_without_denominator_check({ y() - add[0].y3_prev }, (x() - add[0].x3_prev));
+    }
+
+    // Using λ₁, compute x₃ for (A + P):
+    // x₃ = λ₁.λ₁ - x₁ - x
+    Fq x_3 = lambda1.madd(lambda1, { -add[0].x3_prev, -x() });
+
+    // Compute λ₂ for the addition (A + P) + A:
+    // λ₂ = (y - y₃) / (x - x₃)
+    //    = (y - (λ₁ * (x - x₃) - y)) / (x - x₃)    (substituting y₃)
+    //    = (2y) / (x - x₃) - λ₁
+    //
+    x().assert_is_not_equal(x_3);
```

Comment:
This is a fix: added assertion that $x \neq x_3$ where $x$ is the x-coordinate of the original accumulator $Q$ and $x_3$ is the x-coordinate of $(Q + A)$. Not having this check meant that circuit construction would fail if:

$$Q = \pm (Q + A)$$

Case 1: $Q = -(Q+A)$ which implies $A = -2Q$. It is possible for a prover to pass such an input $A$. So we want to disallow this case.

Case 2: $Q = (Q + A)$ which implies $A = \mathcal{O}$. We do not want to allow points at infinity to passed as inputs to the montgomery ladder. Adding the check on the x-coordinates also disallows this case.